### PR TITLE
Rolling dataset generation now truncates dynamic features.

### PR DESCRIPTION
**Pull request by arangatang**
*Issue #, if available:*
Solves #1390 

*Description of changes:*
This PR changes such that dynamic features such as `feat_dynamic_real` and `feat_dynamic_cat` are truncated along with the `target` of the timeseries. 

Previously, running `make_evaluation_predictions` on rolled datasets failed if the dataset had dynamic features. 
This happened as the `target` of the timeseries was shorter than the dynamic features which remained unchanged.  

The changes has been tested with the following code:
```python
import pandas as pd
from gluonts.dataset.common import ListDataset
from gluonts.dataset.field_names import FieldName
from gluonts.dataset.rolling_dataset import (
    generate_rolling_dataset,
    StepStrategy,
)
from gluonts.evaluation.backtest import make_evaluation_predictions
from gluonts.evaluation import Evaluator


prediction_length = 1
train_len = 10
test_len = 15


def create_dataset(length):
    return ListDataset(
        [
            {
                FieldName.TARGET: list(range(length)),
                FieldName.START: pd.Timestamp("01-01-2019"),
                FieldName.FEAT_DYNAMIC_REAL: [list(range(length))],
            }
        ],
        freq="D",
    )


train_ds = create_dataset(train_len)
test_ds = create_dataset(test_len)
rolled_ds = generate_rolling_dataset(
    dataset=test_ds,
    strategy=StepStrategy(prediction_length=1),
    start_time=pd.Timestamp("01-13-2019"),
)

from gluonts.model.deepar import DeepAREstimator
from gluonts.mx.trainer import Trainer

estimator = DeepAREstimator(
    freq="D",
    prediction_length=prediction_length,
    context_length=2 * prediction_length,
    use_feat_dynamic_real=True,
    trainer=Trainer(epochs=1),
)
predictor = estimator.train(training_data=train_ds)
forecast_it, ts_it = make_evaluation_predictions(
    rolled_ds, predictor=predictor, num_samples=100
)
training_agg_metrics, _ = Evaluator(num_workers=0)(ts_it, forecast_it)
from pprint import pprint

pprint(training_agg_metrics)
```
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/gluonts/dataset/rolling_dataset.py [hunk 1]
```diff
@@ -214,6 +227,7 @@ def generate_rolling_dataset(
         for window in strategy.get_windows(prediction_window):
             new_item = item.copy()
             new_item["target"] = np.concatenate([base, window.to_numpy()])
```
Comment (mbohlkeschneider) [inline src/gluonts/dataset/rolling_dataset.py:229]: Can we replace the `target` string with the corresponding `FieldName`?

File: src/gluonts/dataset/rolling_dataset.py [hunk 2]
```diff
@@ -100,6 +100,19 @@ def get_windows(self, window):
             yield window[: int(round(slice_idx))]
 
 
+def truncate_features(timeseries: dict, max_len):
```
Comment (mbohlkeschneider) [inline src/gluonts/dataset/rolling_dataset.py:103]: Maybe add complete type annotations (`max_len` and the return type?)

**Review by mbohlkeschneider — CHANGES_REQUESTED** [review]
Thank you. Do we have a test for that somewhere?

File: src/gluonts/dataset/rolling_dataset.py (hunk 1 above)
Comment (arangatang) [reply src/gluonts/dataset/rolling_dataset.py]: Changed it.

File: src/gluonts/dataset/rolling_dataset.py (hunk 2 above)
Comment (arangatang) [reply src/gluonts/dataset/rolling_dataset.py]: True that, added it!

**Comment by arangatang** [comment]
There is no test in place since previously, i'll add a few.

**Review by mbohlkeschneider — APPROVED** [review]

**Comment by mbohlkeschneider** [comment]
Docs build is failing. Can you fix that @arangatang ?

**Comment by arangatang** [comment]
Yeah I started to look into it yesterday but I was not able to reproduce it locally. 
There seem to be some issue with the `synthetic_data_generation.md` notebook in the docs folder. It cannot be built since  #1380 although the changes made there seems unrelated. 
Perhaps there is a dependency which was updated somewhere, I'll keep digging!

**Pull request merged**